sidechain/consensus/common/src/block_import_confirmation_handler.rs

```diff
@@ -0,0 +1,122 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::error::{Error, Result};
+use itc_parentchain_light_client::{
+	concurrent_access::ValidatorAccess, BlockNumberOps, ExtrinsicSender, NumberFor,
+};
+use itp_extrinsics_factory::CreateExtrinsics;
+use itp_node_api_metadata::pallet_sidechain::SidechainCallIndexes;
+use itp_node_api_metadata_provider::AccessNodeMetadata;
+use itp_types::{OpaqueCall, ShardIdentifier};
+use log::*;
+use sidechain_primitives::traits::Header as HeaderTrait;
+use sp_runtime::traits::Block as ParentchainBlockTrait;
+use std::{marker::PhantomData, sync::Arc};
+
+/// Trait to confirm a sidechain block import. Typically this is sent to the parentchain.
+pub trait ConfirmBlockImport<SidechainHeader> {
```

Comment:
```suggestion
pub trait ConfirmSidechainBlockImport<SidechainHeader> {
```

Reply:
Should we really do this? The trait lives only in the sidechain-consensus code. So it is clear that this is about the sidechain (and not some other chain). ... I think long names make the code more difficult to read and should be avoided if possible.

Reply:
Long names are difficult to read, agreed. But two things are pro:
+ Makes it easier to use the search function, since we have many "BlockSomething" names.
+ The sidechain also confirms parentchain block imports. It lives somewhere else, but is it really that obvious?

I'll leave the decision up to you. Both work for me.

Reply:
I think long names are justified if they avoid ambiguity and are accurate. However, in this case, I would agree with @masapr - this type is inside the `sidechain` crate, which makes it unambiguous. Most of our types inside the sidechain folder, don't have the `Sidechain` prefix. Only when importing in a place where we have both, sidechain and parentchain types (like `enclave-runtime`), we add the prefixes to distinguish the types.